src/render.rs

```diff
@@ -356,6 +439,14 @@ impl<'a> Renderer<'a> {
 
                 Ok(output)
             },
+            ImportMacro {tpl_name, name} => {
+                let tpl = try!(self.tera.get_template(&tpl_name));
+                self.macros.insert(name.to_string(), tpl.macros.clone());
+                // In theory, the render_node should return TeraResult<Option<String>>
+                // but in practice there's no difference so keeping this hack
```

Comment:
Except this allocates needlessly. Just do `let x = match node` at the top. Here, `return Ok(None)`. Everywhere else, remove the `Ok(..)`, but keep what's inside. Then add a final `Ok(x)` to the end of the function.